### PR TITLE
Added converter from gogo proto to google proto.

### PR DESCRIPTION
**Pull request by cristure**

### CODE REVIEW

File: data/type.go [hunk 1]
```diff
@@ -0,0 +1,27 @@
+package data
+
+type FieldsGetter interface {
```
Comment (ssd04) [inline data/type.go:3]: please add comments for these

File: process/outportBlockConverter.go [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+	"github.com/multiversx/mx-chain-core-go/data/outport"
+	"github.com/multiversx/mx-chain-core-go/marshal"
+	"google.golang.org/protobuf/proto"
+
+	"github.com/multiversx/mx-chain-ws-connector-template-go/data"
+)
+
+type outportBlockConverter struct {
+	gogoProtoMarshalizer *marshal.GogoProtoMarshalizer
+	bigIntCaster         coreData.BigIntCaster
```
Comment (ssd04) [inline process/outportBlockConverter.go:19]: we can have here a separate component for standard proto marshaller, which implements marshal.Marshaller from core

File: process/protoMarshalizer.go [hunk 3]
```diff
@@ -0,0 +1,29 @@
+package process
+
+import (
+	"fmt"
+
+	"google.golang.org/protobuf/proto"
+)
+
+type ProtoMarshalizer struct{}
```
Comment (ssd04) [inline process/protoMarshalizer.go:9]: missing comments

File: process/protoMarshalizer_test.go [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+			}
+		}
+	}()
+	err = protoMarshaller.Unmarshal(obj, buf)
+	return
+}
+
+func TestGogoProtoMarshalizer_Marshal(t *testing.T) {
+	outportBlock := data.ShardOutportBlock{}
+	encNode, err := recovedMarshal(&outportBlock)
```
Comment (ssd04) [inline process/protoMarshalizer_test.go:47]: update test names, remove Gogo

Comment (ssd04) [reply process/protoMarshalizer_test.go:47]: add parallel tests run

**Review by ssd04 — APPROVED** [review]

File: process/outportBlockConverter.go [hunk 5]
```diff
@@ -0,0 +1,174 @@
...
+		return nil, fmt.Errorf("cannot convert to meta outport block. header type: %s not supported", outportBlock.BlockData.HeaderType)
+	}
+
+	// marshal with gogo, since the outportBlock is gogo protobuf (coming from the node).
+	bytes, err := o.gogoProtoMarshalizer.Marshal(outportBlock)
+	if err != nil {
+		return nil, fmt.Errorf("marshal metaBlockCaster error: %w", err)
+	}
+
+	// unmarshall into google protobuf. This is the proto that will be used in firehose.
```
Comment (AdoAdoAdo) [inline process/outportBlockConverter.go:159]: can you remove the reference to firehose in the comment? the converter will be used also for other RPC integrations

File: data/fireOutport.proto [hunk 6]
```diff
@@ -0,0 +1,438 @@
+syntax = "proto3";
+
+package proto;
+
+option go_package = "github.com/multiversx/mx-chain-ws-connector-template-go/data";
+
+message HyperBlock {
```
[the remark this answers is not in the document]
Comment (cristure) [reply data/fireOutport.proto:7]: yes, that's true. they will come in the gRPC server PR.

File: data/fireOutport.proto [hunk 7]
```diff
@@ -0,0 +1,438 @@
+syntax = "proto3";
```
[the remark this answers is not in the document]
Comment (cristure) [reply data/fireOutport.proto:1]: this is done in the gRPC PR. can we proceed with this one as it is?

File: process/protoMarshalizer.go [hunk 8]
```diff
@@ -0,0 +1,32 @@
...
+// Unmarshal a slice of bytes into a proto message.
+func (p *ProtoMarshalizer) Unmarshal(obj interface{}, buff []byte) error {
+	message, ok := obj.(proto.Message)
+	if !ok {
+		return fmt.Errorf("cannot marshal %T into a proto.Message", obj)
+	}
+	return proto.Unmarshal(buff, message)
+}
+
+func (p *ProtoMarshalizer) IsInterfaceNil() bool {
```
[the remark this answers is not in the document]
Comment (cristure) [reply process/protoMarshalizer.go:30]: added.

File: process/protoMarshalizer_test.go [hunk 9]
```diff
@@ -0,0 +1,86 @@
...
+	assert.Nil(t, err)
+	assert.NotNil(t, encNode)
+}
+
+func TestProtoMarshalizer_MarshalWrongObj(t *testing.T) {
+	t.Parallel()
+
+	obj := "multiversx"
+	encNode, err := recovedMarshal(obj)
```
[the remark this answers is not in the document]
Comment (cristure) [reply process/protoMarshalizer_test.go:58]: I am trying to marshal a string into a proto.Message. It should fail. You are right an assertion on the error would be better.

File: process/outportBlockConverter.go (hunk 5 above)
Comment (cristure) [reply process/outportBlockConverter.go:159]: changed.

**Review by AdoAdoAdo — APPROVED** [review]

**Pull request merged**